omegatrade-app/frontend/src/app/components/company/update-company/update-company.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { FormBuilder, Validators } from '@angular/forms';
+import { RestService } from '../../../services/rest.service';
+import { SnackBarService } from '../../../services/snackbar.service';
+
+@Component({
+  selector: 'app-update-company',
+  templateUrl: './update-company.component.html',
+  styleUrls: ['./update-company.component.css']
+})
+export class UpdateCompanyComponent implements OnInit {
+  companyForm: any;
+  loader: boolean = false;
+  constructor(private snackBarService: SnackBarService, private restService: RestService, private formBuilder: FormBuilder, private dialogRef: MatDialogRef<UpdateCompanyComponent>, @Inject(MAT_DIALOG_DATA) public data: any) {
+    this.companyForm = this.formBuilder.group({
+      companyId: ['', []],
+      companyName: ['', [Validators.required]],
+      companyShortCode: ['', [Validators.required]],
+      created_at: ['']
+    });
+    if (data && data.companyId) {
+      this.companyForm.setValue({
+        companyId: data.companyId,
+        companyName: data.companyName,
+        companyShortCode: data.companyShortCode,
+        created_at: data.created_at
+      });
+      const ctrl = this.companyForm.get('companyShortCode');
+      ctrl.disable();
+    }
+  }
+
+  submitForm() {
+    if (this.companyForm.dirty && this.companyForm.valid) {
+      this.loader = true;
+      const path = (this.companyForm.get('companyId').value != '') ? 'update' : 'create';
```

Comment:
We can provided input text for the company code and we disabled it for the EDIT case, so for the add company case it allow user to provide, so this condition always be true and so path variable will contain always 'update', please check it once

Reply:
we are handling Create and Update component with a single component. 
and checking with `companyId` in order to update the path whether the post function should perform update or create.
To create a company, companyId will always null.